nimbus/src/enrollment.rs

```diff
@@ -1,9 +1,9 @@
 // This Source Code Form is subject to the terms of the Mozilla Public
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
```

Comment:
This file was the one that seemed the most in need of some principled opinions - the functions in here variously take or do not take a `Writer` depending on how other parts of the code expect to call them. I've made it so that *none* of these functions are responsible for managing their own transactions.